linker/Mono.Linker/DirectoryAssemblyResolver.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.IO;
+using Mono.Collections.Generic;
+using Mono.Cecil;
+
+#if NET_CORE
+namespace Mono.Linker {
+
+	public sealed class AssemblyResolutionException : FileNotFoundException {
+
+		readonly AssemblyNameReference reference;
+
+		public AssemblyNameReference AssemblyReference {
+			get { return reference; }
+		}
+
+		public AssemblyResolutionException (AssemblyNameReference reference)
+			: base (string.Format ("Failed to resolve assembly: '{0}'", reference))
+		{
+			this.reference = reference;
+		}
+	}
+
```

Comment:
Sure, we can do that. I opened a PR on cecil: https://github.com/jbevain/cecil/pull/340. I'll update the submodule when it gets merged.